spec/source_file_spec.rb

```diff
@@ -59,11 +63,40 @@
       it "has 80% covered_percent" do
         expect(subject.covered_percent).to eq(80.0)
       end
+
+      it "Has total branches count 4" do
+        expect(subject.total_branches.size).to eq(4)
+      end
+
+      it "Has covered branches count 2" do
+        expect(subject.covered_branches.size).to eq(2)
+      end
+
+      it "Has missed branches count 2" do
+        expect(subject.missed_branches.size).to eq(2)
+      end
+
+      it "Has root branches count 2" do
+        expect(subject.root_branches.size).to eq(2)
+      end
+
+      it "Has branch on line number 7 with report pr line" do
+        expect(subject.branch_per_line(17)).to eq("[3, \"+\"]")
+      end
+
+      it "Has coverage report" do
+        expect(subject.branches_report).to eq(17 => [[3, "+"]], 19 => [[0, "-"]], 29 => [[3, "+"]], 31 => [[0, "-"]])
```

Comment:
19 and 31 should be 20 and 32? See earlier comment about zero-based keys.

Reply:
this case is a multi-line branch `if, else..`, these branches should have badges on the line of the condition definition, that's why I have done `branch.start_line - 1`, otherwise, the badge statistics would appear on the first line of the branch body which is misleading from my point of view.